File: DocSamples/WebFormWidgetsSamples/HelloWorld/HelloWorldDesigner.js
//Documentation articles: https://docs.sitefinity.com/link-widget-components
Type.registerNamespace("SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorld");

SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorldDesigner = function (element) {
    SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorldDesigner.initializeBase(this, [element]);
};

SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorldDesigner.prototype = {
    initialize: function () {
        SitefinityWebApp.Widgets.HelloWorldDesigner.callBaseMethod(this, 'initialize');
    },
    dispose: function () {
        SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorldDesigner.callBaseMethod(this, 'dispose');
    },
    refreshUI: function () {
        var controlData = this._propertyEditor.get_control();

        // bind widget properites to designer
        jQuery("#txtName").val(controlData.Name);
    },
    applyChanges: function () {

        var controlData = this._propertyEditor.get_control();

        // bind designer properties back to widget
        controlData.Name = jQuery("#txtName").val();
    }
};

SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorldDesigner.registerClass('SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorldDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);

File: DocSamples/WebFormWidgetsSamples/HelloWorld/HelloWorldDesigner.cs
//Documentation articles: https://docs.sitefinity.com/link-widget-components

using System.Collections.Generic;
using System.Web.UI;
using Telerik.Sitefinity.Web.UI.ControlDesign;

namespace SitefinityWebApp.WebFormWidgetsSamples.HelloWorld
{
    public class HelloWorldDesigner : ControlDesignerBase
    {
        protected override void InitializeControls(Telerik.Sitefinity.Web.UI.GenericContainer container)
        {
            base.DesignerMode = ControlDesignerModes.Simple;
        }

        //Link the designer to its template

        public override string LayoutTemplatePath
        {
            get
            {
                if (string.IsNullOrEmpty(base.LayoutTemplatePath))
                    return HelloWorldDesigner.layoutTemplatePath;
                return base.LayoutTemplatePath;
            }
            set
            {
                base.LayoutTemplatePath = value;
            }
        }
        private static readonly string layoutTemplatePath = "SitefinityWebApp.WebFormWidgetsSamples.HelloWorld.HelloWorldDesigner.ascx";

        //Register the control designer script

        public override IEnumerable<ScriptReference> GetScriptReferences()
        {
            // get script collection
            var scripts = base.GetScriptReferences() as List<ScriptReference>;
            if (scripts == null) return base.GetScriptReferences();

            scripts.Add(new ScriptReference(scriptReference));

            return scripts.ToArray();
        }
        private string scriptReference = "~/WebFormWidgetsSamples/HelloWorld/HelloWorldDesigner.js";
    }
}